### PR TITLE
Implement meshing for `Capsule2d`

### PR DESCRIPTION
**Pull request by Jondolf**
# Objective

The `Capsule2d` primitive was added in #11585. It should support meshing like the other 2D primitives.

## Solution

Implement meshing for `Capsule2d`.

It doesn't currently support "rings" like Bevy's `Capsule` shape (not `Capsule3d`), but it does support resolution to control the number of vertices used for one hemicircle. The total vertex count is two times the resolution; if we allowed setting the full vertex count, odd numbers would lead to uneven vertex counts for the top and bottom hemicircles and produce potentially unwanted results.

The capsule looks like this (with UV visualization and wireframe) using resolutions of 16, 8, and 3:

Resolution 16
Resolution 8
Resolution of 3

The `2d_shapes` example now includes the capsule, so we also get one more color of the rainbow 🌈

New 2D shapes example

### CODE REVIEW

File: crates/bevy_render/src/mesh/primitives/dim2.rs [hunk 1]
```diff
@@ -266,3 +266,142 @@ impl From<Rectangle> for Mesh {
...
+    }
+
+    /// Builds a [`Mesh`] based on the configuration in `self`.
+    pub fn build(&self) -> Mesh {
+        // The resolution is the number of vertices for one semicircle
+        let resolution = self.resolution as u32;
+        let vertex_count = 2 * self.resolution;
+
+        // Six extra indices for the two triangles between the hemicircles
+        let mut indices = Vec::with_capacity((vertex_count - 2) * 3 + 6);
```
Comment (rparrett) [inline crates/bevy_render/src/mesh/primitives/dim2.rs:317]: Not sure where the error is, but this seems wrong. The default capsule is initially given a capacity of 96, but the length at the end is 90.

Comment (rparrett) [reply crates/bevy_render/src/mesh/primitives/dim2.rs:317]: I believe this should be `(resolution - 2) * 2 * 3 + 6`.

Comment (Jondolf) [reply crates/bevy_render/src/mesh/primitives/dim2.rs]: Yep, you're correct! I tested with different resolutions to be sure, and that works perfectly

**Review by rparrett — APPROVED** [review]
nit: semicircle/hemicircle and half-height/half-length are used pretty interchangeably throughout docs/comments and it would be nice to standardize a bit.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**